src/components/Client/Client.js

```diff
@@ -0,0 +1,157 @@
+const axios = require('axios');
+
+class GraphQLClient {
+
+    /** The GraphQLClient retrieves data from the FAIRSharing API and sends it to the front-end.
+     * Be careful, this is a singleton and trying to cast new instances will return the existing instance. Be
+     * also careful, its constructor is async !!
+     * @returns {Promise} - to use this object you need to do "await new ClassName()" or use .then(callback)
+     */
+    constructor(){
+        if (GraphQLClient._instance){
+            return GraphQLClient._instance
+        }
+        GraphQLClient._instance = this;
+        this.url = "https://api.fairsharing.org/graphql";
+        this.headers = {
+            "Accept": "application/json",
+            "Content-Type": "application/json"
+        };
+    }
+
+    /**
+     * Execute the given query (coming from a json file, see /queries/getRecords.json)
+     * @param {Object} query - the query coming from the JSON file
+     * sending to the API.
+     * @returns {Promise}
+     */
+    async executeQuery(query){
+
+        let client = this;
+
+        let queryString = this.queryBuilder({
+                fields: query.queryFields.target.fields,
+                pagination: query.pagination,
+                queryName: query.queryName,
+                objectType: query.queryFields.target.name,
+                queryFields: query.queryFields["elasticSearchFields"].join(" ") + " ",
+                queryParam: query["queryParam"]
+            });
+
+        // trigger the query
+        try {
+            let resp = await axios.post(this.url, queryString, this.headers);
+            if (resp.data.errors){
+                throw new Error(resp.data.errors[0].message);
+            }
+
+            let content = resp.data.data;
+
+            /* group By */
+            query.queryFields.target.fields.forEach(function(queryField){
+                if (queryField.type === "object"){
+                    queryField.target.forEach(function(target){
+                        if (queryField.hasOwnProperty(target) && queryField[target].hasOwnProperty("groupBy")){
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Do not access Object.prototype method 'hasOwnProperty' from target object.](https://app.codacy.com/gh/FAIRsharing/fairsharing.github.io/pullRequest?prid=4893952)